config/config.go

```diff
@@ -37,26 +51,31 @@ func parseTerragruntConfigFile(configPath string) (*TerragruntConfig, error) {
 }
 
 // Parse the Terragrunt config contained in the given string
-func parseTerragruntConfig(config string) (*TerragruntConfig, error) {
-	terragruntConfig := &TerragruntConfig{}
-
-	if err := hcl.Decode(terragruntConfig, config); err != nil {
+func parseConfigString(configSrc string) (*Config, error) {
+	f := &fileConfig{}
```

Comment:
Same comment on naming here too. You're using `f` here to mean something completely different than the `f` in the `lookupLock` function above. To understand what this is, I'm forced to scroll and read a lot of code. With a clearer name like `terragruntConfigFromFile`, I can guess it from just the name. This makes it not only easier to read the code, but also prevents errors (see [Making Wrong Code Look Wrong](http://www.joelonsoftware.com/articles/Wrong.html)).